sec_certs_page/user/views.py

```diff
@@ -0,0 +1,464 @@
+from flask_mail import Message
+
+from .. import mongo, mail
+from ..common.views import register_breadcrumb
+from ..common.permissions import admin_permission
+from ..notifications.utils import derive_token
+from . import user
+from .forms import LoginForm, RegisterForm, PasswordResetRequestForm, PasswordResetForm, MagicLinkForm
+from .models import User
+
+# OAuth support (optional - only if GitHub OAuth is configured)
+try:
+    from flask_dance.contrib.github import github
+    oauth_available = True
+except ImportError:
+    oauth_available = False
+
+
+@user.route("/login", methods=["GET", "POST"])
+@register_breadcrumb(user, ".login", "Login")
+def login():
+    form = LoginForm()
+    if form.is_submitted():
+        if form.validate():
+            user_obj = User.get(form.username.data)
+            if user_obj and user_obj.check_password(form.password.data):
+                login_user(user_obj, form.remember_me.data)
+                identity_changed.send(current_app._get_current_object(), identity=Identity(user_obj.id))
+                flash("You've been successfully logged in.", "info")
+                if admin_permission.can():
+                    return redirect(url_for("admin.index"))
+                else:
+                    return redirect(url_for("index"))
+            else:
+                flash("Bad.", "error")
+    return render_template("user/login.html.jinja2", form=form)
+
+
+@user.route("/logout")
+@login_required
+def logout():
+    logout_user()
+    for key in ("identity.name", "identity.auth_type"):
+        session.pop(key, None)
+    identity_changed.send(current_app._get_current_object(), identity=AnonymousIdentity())
+    flash("You've been successfully logged out.", "info")
+    return redirect(url_for("index"))
+
+
+@user.route("/register", methods=["GET", "POST"])
+@register_breadcrumb(user, ".register", "Register")
+def register():
+    if current_user.is_authenticated:
+        return redirect(url_for("index"))
+    
+    form = RegisterForm()
+    if form.validate_on_submit():
+        # Create user
+        try:
+            new_user = User.create(
+                username=form.username.data,
+                email=form.email.data,
+                password=form.password.data,
+                roles=[]
+            )
+            # Send confirmation email
+            token = User.generate_confirmation_token(new_user.username)
+            confirmation_url = url_for('user.confirm_email', token=token, _external=True)
+            
+            msg = Message(
+                subject="Confirm your email - sec-certs",
+                recipients=[new_user.email],
+                html=render_template("user/emails/confirm_email.html.jinja2", 
+                                   user=new_user, confirmation_url=confirmation_url),
+                sender=current_app.config.get('MAIL_DEFAULT_SENDER')
+            )
+            
+            try:
+                mail.send(msg)
```

Comment:
This and all of the other email sends should be handled asynchronously via dramatiq tasks. Just like how the notification emails are handled in the notifications blueprint.